### PR TITLE
Support for pyproject.toml

### PR DESCRIPTION
**Pull request by girip11**
This is an optional PR. This tries to address https://github.com/nbQA-dev/nbQA/issues/211. **You are free to disapprove/reject this PR completely.** 

This makes toml as optional dependency. If toml is not present, then it won't parse from the pyproject.toml.

Since you mentioned in the issue

>That's a good suggestion, my only worry is that it would increase complexity, while having just a single file in which to look for configs would be easier to maintain...will think about it

I refactored the config part, I found it easier to add support to pyproject.toml with minimal code changes. So I went ahead and added those changes. I you want to support only `setup.cfg`, it's completely fine with me.

### CODE REVIEW

**Comment by girip11** [comment]
As @s-weigand  suggested, I looked at how coverage implements toml as optional dependency and used the idea here.

File: docs/configuration.rst [hunk 1]
```diff
@@ -63,9 +87,16 @@ To ignore extra cells, you can use the :code:`--nbqa-ignore-cells` CLI argument,
...
 
     [nbqa.ignore_cells]
     black = %%html,%%cython
+
+, or the :code:`ignore_cells` option in your :code:`pyproject.toml` file, e.g.
+
+.. code-block:: toml
+
+    [tool.nbqa.ignore_cells]
+    black = "%%html,%%cython"
```
Comment (MarcoGorelli) [inline docs/configuration.rst:102]: Wow, awesome!!

Does `toml` require the `%` to be escaped like `cfg` does? (I've opened #256 for the latter, which I forgot about when writing it :flushed: )

Comment (girip11) [reply docs/configuration.rst:102]: Thanks for pointing it out. I have no idea on this. Let me read about it and will address this comment accordingly.

Comment (girip11) [reply docs/configuration.rst:102]: For TOML, we don't need to escape `%`. I have added a test case to confirm the same.

**Comment by MarcoGorelli** [comment]
> For TOML, we don't need to escape %. I have added a test case to confirm the same.

Wow, that's awesome! Thanks for looking into this. OK, I think that `pyproject.toml` should then be the default configuration file, as having to escape `%` is confusing. We can add `toml` as a required dependency, as it's a really small package anyway.

So, my suggestion would be:
- remove references to `setup.cfg` in the docs, and just have `pyproject.toml` as an example. I think that's easier than saying "otherwise, you can use setup.cfg, but you'll have to escape the percent signs"
- change `setup.py` so that `toml` is a required dependency

**Comment by girip11** [comment]
Sure. I will make those changes and update the PR. 

>  OK, I think that pyproject.toml should then be the default configuration file, as having to escape % is confusing.

Support for `setup.cfg` and `tox.ini` would still be present in the code in addition to `pyproject.toml` right? Just that we would update the documentation to use pyproject.toml. Is my understanding correct?

**Comment by MarcoGorelli** [comment]
Yes, if you agree I think we can still silently support them, but only list `pyproject.toml` in the docs

**Comment by girip11** [comment]
Sure. We will support those other files. I will make the required changes.

File: docs/configuration.rst [hunk 2]
```diff
@@ -12,28 +12,31 @@ If you wish to pass extra flags (e.g. :code:`--ignore W503` to :code:`flake8`) y
...
+    ]
 
-    [nbqa.addopts]
-    flake8 = --ignore W503
 
 Config file
 ~~~~~~~~~~~
 
-By default, :code:`nbQA` will use :code:`setup.cfg` to get the configs for your third-party tool.
+By default, :code:`nbQA` will look up the configuration of the third party tool in the following files :code:`setup.cfg`, :code:`tox.ini` and :code:`pyproject.toml`.
```
Comment (MarcoGorelli) [inline docs/configuration.rst:28]: I think we can just mention `pyproject.toml` here, if it's now a required dependency

File: nbqa/toml_parser.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+def parse_from_pyproject_toml(command: str, file_path: Path) -> Optional[Configs]:
+    """
+    Parse nbqa configuration from TOML file.
+
+    Parameters
+    ----------
+    command : str
+        Third party tool to run
+    config_sections : List[str]
+        Configuration sections supported by nbqa e.g addopts, config, mutate
```
Comment (MarcoGorelli) [inline nbqa/toml_parser.py:22]: missing parameter?

File: tox.ini [hunk 4]
```diff
@@ -2,7 +2,10 @@
 envlist = py36, py37, py38
 
 [testenv]
-deps=-r requirements-dev.txt
+extras =
+    toml
```
Comment (MarcoGorelli) [inline tox.ini:6]: Is this still necessary?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Awesome, have just left some minor comments

File: docs/configuration.rst (hunk 2 above)
Comment (MarcoGorelli) [reply docs/configuration.rst:28]: nevermind me, this comment is correct, sorry :innocent:

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks a ton!

**Pull request merged**